ipam/allocator.go

```diff
@@ -374,13 +380,30 @@ func (a *Allocator) retrieveBitmask(k SubnetKey, n *net.IPNet) (*bitseq.Handle,
 func (a *Allocator) getPredefineds(as string) []*net.IPNet {
 	a.Lock()
 	defer a.Unlock()
-	l := make([]*net.IPNet, 0, len(a.predefined[as]))
-	for _, pool := range a.predefined[as] {
-		l = append(l, pool)
+
+	p := a.predefined[as]
```

Comment:
I like the use of append.  Will incorporate.   But don't want to use `%` because the list could, in theory, be empty leading to a divide-by-zero exception.

Reply:
I think that safety check can be added as if before these 3 lines. It just keeps the code simple and short.